core/src/main/java/io/questdb/cairo/wal/seq/TableSequencerAPI.java

```diff
@@ -72,30 +73,35 @@ public void close() {
         releaseAll();
     }
 
-    public void forAllWalTables(final RegisteredTable callback) {
+    public void forAllWalTables(RegisteredTable callback) {
         final CharSequence root = configuration.getRoot();
         final FilesFacade ff = configuration.getFilesFacade();
 
         // this will be replaced with table name registry when drop WAL table implemented
         try (Path path = new Path().of(root).slash$()) {
+            final LastWalTxnReader walTxnReader = new LastWalTxnReader(configuration.getFilesFacade());
+            final WalTableInfo info = new WalTableInfo();
             final StringSink nameSink = new StringSink();
             int rootLen = path.length();
             ff.iterateDir(path, (name, type) -> {
                 if (Files.isDir(name, type, nameSink)) {
                     path.trimTo(rootLen);
                     if (isWalTable(nameSink, path, ff)) {
-                        long lastTxn;
-                        int tableId;
-                        try (TableSequencer tableSequencer = openSequencerLocked(nameSink, SequencerLockType.NONE)) {
-                            lastTxn = tableSequencer.lastTxn();
-                            tableId = tableSequencer.getTableId();
+                        path.trimTo(rootLen);
+                        try {
+                            readWalTableInfo(nameSink, info, walTxnReader, path);
                         } catch (CairoException ex) {
-                            LOG.critical().$("could not open table sequencer [table=").$(nameSink).$(", errno=").$(ex.getErrno())
+                            LOG.critical().$("could not read WAL table metadata [table=").$(nameSink).$(", errno=").$(ex.getErrno())
```

Comment:
```suggestion
                            LOG.critical().$("could not read WAL table metadata [table=").utf8(nameSink).$(", errno=").$(ex.getErrno())
```

Reply:
Done in fd5a963253d0d9ea64055cee75eb555bab5d8cae